src/operations.jl

```diff
@@ -22,10 +27,15 @@ function transform(src::T, funcs::F) where {T, F}
     return Transforms{C, typeof(x), F}(x, funcs)
 end
 
-getfunc(row, nt::NamedTuple, nm) = get(nt, nm, identity)
-getfunc(row, d::Dict{String, <:Base.Callable}, nm) = get(d, String(nm), identity)
-getfunc(row, d::Dict{Symbol, <:Base.Callable}, nm) = get(d, nm, identity)
-getfunc(row, d::Dict{Int, <:Base.Callable}, nm) = get(d, findfirst(isequal(nm), propertynames(row)), identity)
+getfunc(row, nt::NamedTuple, nm::Symbol) = get(nt, nm, identity)
+getfunc(row, d::Dict{String, <:Base.Callable}, nm::Symbol) = get(d, String(nm), identity)
+getfunc(row, d::Dict{Symbol, <:Base.Callable}, nm::Symbol) = get(d, nm, identity)
+getfunc(row, d::Dict{Int, <:Base.Callable}, nm::Symbol) = get(d, findfirst(isequal(nm), columnnames(row)), identity)
+
+getfunc(row, nt::NamedTuple, i::Int) = i > fieldcount(typeof(nt)) ? identity : getfield(nt, i)
+getfunc(row, d::Dict{String, <:Base.Callable}, i::Int) = get(d, String(columnnames(row)[i]), identity)
+getfunc(row, d::Dict{Symbol, <:Base.Callable}, i::Int) = get(d, columnnames(row)[i], identity)
+getfunc(row, d::Dict{Int, <:Base.Callable}, i::Int) = get(d, i, identity)
```

Comment:
Can we drop the dispatch on `Base.Callable` ?

Reply:
Does it hurt at all? Or overly restrict?

Reply:
It's overly restrictive.
Doesn't work with functors.

Maybe I want to pass something though a trained NN. `Flux.Chain` is callable, but not a subtype of `Base.Callaeble`